src/main/java/services/VmMdevService.java

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright (c) 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services;
+
+import annotations.Area;
+import mixins.Follow;
+import org.ovirt.api.metamodel.annotations.In;
+import org.ovirt.api.metamodel.annotations.Out;
+import org.ovirt.api.metamodel.annotations.Service;
+import types.VmMdev;
+
+@Service
+@Area("Virtualization")
+public interface VmMdevService {
+
+    /**
+     * Retrieves the configuration of mediated devices in the virtual machine.
+     *
+     * @author Milan Zamazal <mzamazal@redhat.com>
+     * @date 10 March 2022
+     * @status added
+     * @since 4.5
+     */
+    interface Get extends Follow {
+        /**
+         * The information about the mediated device of the virtual machine.
+         *
+         * @author Milan Zamazal <mzamazal@redhat.com>
+         * @date 10 Mar 2022
+         * @status added
+         * @since 4.5
+         */
+        @Out VmMdev mdev();
+
+        /**
+         * Specifies if the data returned should correspond to the next execution of
+         * the virtual machine, or to the current execution.
+         *
+         * For example, to get data for the current execution of the virtual machine,
+         * send a request like this:
+         *
+         * [source]
+         * ----
+         * GET /ovit-engine/api/vms/123/mdevs/456?current=true
+         * ----
+         *
+         * The default value is `false`.
+         *
+         * @author Milan Zamazal <mzamazal@redhat.com>
+         * @date 10 Mar 2022
+         * @status added
+         * @since 4.5
+         */
+        @In Boolean current();
+    }
+
+    /**
+     * Remove the mediated device from the virtual machine.
+     *
+     * @author Milan Zamazal <mzamazal@redhat.com>
+     * @date 10 March 2022
+     * @status added
+     * @since 4.5
+     */
+    interface Remove {
+        /**
+         * Indicates if the remove should be performed asynchronously.
+         */
+        @In Boolean async();
+    }
+}
```

Comment:
shouldn't we also have an 'update' method in order to change their spec-params? (or clients would need to remove the device and add it with updated spec-params?)

Reply:
Update methods were added, to have the API complete, although they are not strictly necessary because there is only single item to change (specParams).  So there is no added value in comparison to remove+add, but it at least saves one API call.